nodestream_akamai/edns/edns.py

```diff
@@ -0,0 +1,67 @@
+import logging
+from ipaddress import IPv4Address, ip_address
+
+from nodestream.pipeline.extractors import Extractor
+
+from ..akamai_utils.edns_client import AkamaiEdnsClient
+
+
+class AkamaiEdnsExtractor(Extractor):
+    def __init__(self, **akamai_client_kwargs) -> None:
+        self.client = AkamaiEdnsClient(**akamai_client_kwargs)
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+    def get_address_format(self, address):
+        try:
+            if type(ip_address(address)) is IPv4Address:
+                return "ipv4"
+            else:
+                return "ipv6"
+        except ValueError:
+            if " " in address:
+                return "Invalid"
+            else:
+                return "endpoint"
+
+    async def extract_records(self):
+        supported_record_types = ["A", "AAAA", "CNAME", "NS", "CAA"]
+        for zone in self.client.list_zones():
+            try:
+                zone["recordsets"] = self.client.list_recordsets(zone["zone"])
+            except Exception:
+                self.logger.error(
+                    f"Failed to list record sets for zone: {zone['zone']}"
+                )
+            for i in range(len(zone["recordsets"])):
```

Comment:
Is `i` necessary here? Could this be done with a comprehension instead?

```suggestion
            zone["recordsets"] = [_modify_recordset(rs, zone["zone"]) for rs in zone["recordsets"]]
```

I'm not sure this is a huge change/improvement, but it has a side effect of reducing the total amount of indentation to keep track of.

in my brain, the fn would look something like...
```python
def _modify_recordset(zone_recordset, zone):
    recordset_type = zone_recordset["type"]
    if recordset_type in SUPPORTED_RECORD_TYPES:
        zone_recordset["key"] = zone_recordset["name"] + "/" + recordset_type
        zone_recordset["zone"] = zone["zone"]
        ### 
        # if-blocks for each supported type
        ###
    return zone_recordset
```

Reply:
totally cool if this is to manage memory usage for arbitrarily long lists of record-sets.  The comprehensions idea will end up creating a whole new list, which might be too much to stuff into memory.

Reply:
Don't want to split out to a new function which is only used once, as it makes the code harder to read, and the indentation level isn't egregious just yet. The i iterator is needed as I don't know of a way to update the source object in a standard for or comprehension, but if you have a technique I am all ears